rfc/rfc-92/rfc-92.md

```diff
@@ -0,0 +1,200 @@
+4. It's very standard practice in databases to allow multiple storage backends (MySQL supports myISAM, innodb/btree, myrocks/lsm). This may be crucial step towards the database northstar vision.
+5. As a long time member of the Hudi community and open-source enthusiast, I think supporting other table formats, even existing ones like Apache Iceberg or Delta Lake, benefits those communities as well.
+    1. For e.g. the Hudi Streamer tool being used at our data lake (and hundreds more) for ingestion/incremental ETL can also benefit other communities.
+    2. Hudi provides out-of-box automatic table management that is manual in projects like Iceberg. With such an implementation, common data lake services can be reused across formats.
+    3. Hudi's high performance writer path can be extended to other formats (to the extent possible, that is not dependent on features like indexing that is only in Hudi's native format)
+    4. there are more such services and functionalities to be unlocked.
+
+
+Some non-technical reasons:
+1. Though Hudi is clearly defined as a platform over the years, there is so much vendor attention in the space for the past couple years, where Hudi is minimized to a table format and compared. This change will help highlight the value of Hudi's open software services, beyond just open formats.
+2. It may be controversial to say this. But, the project has been facing a lot of vendor FUD due to different vendors supporting different table formats. It is neither in the interest nor the business of the project community to be part of vendor wars. Opening up the table format layer to different implementations avoids these distractions for regular OSS contributors with no vendor interests, and helps focus on open-source software design and development.
+
+  
+## **Implementation**
+
+The main implementation step here is to create abstraction called TableFormatPlugin which handles table format operations such as 
+
+
+1. Committing writes 
+2. Update and read table format specific Metadata operations 
+3. Timeline 
+4. Conflict Resolution
+5. Lock Provider
+6. Rollbacks
+
+
+The Hudi platform is responsible for managing the data path and can be configured with the table format plugin by default using Hudi native table format. Other table formats will have their own implementation of this abstraction.  
+
+### Commit Protocol:
+
+Hudi uses the creation of a .commit file to advertise the completion of a commit on the table. With the TableFormatPlugin, this will be a two step process:
+
+1.  Creation of .commit file in .hoodie timeline 
+2. Store the commit completion time in the table format's commit metadata.
+
+  
+With this, the write is only completed when both the above steps are completed. The plugin provided timeline needs to fence the timeline ensuring the definition of complete stays consistent.  This ensures the snapshot isolation is maintained.
```

Comment:
Need to call out that - all Hudi writers need to match the table format being used? Do we need a table property

Reply:
Yes. Added.